stacks-signer/src/client/stackerdb.rs

```diff
@@ -190,15 +191,11 @@ impl<M: MessageSlotID + 'static> StackerDB<M> {
             let send_request = || session.put_chunk(&chunk).map_err(backoff::Error::transient);
             let chunk_ack: StackerDBChunkAckData = retry_with_exponential_backoff(send_request)?;
 
-            if let Some(versions) = self.slot_versions.get_mut(msg_id) {
-                // NOTE: per the above, this is always executed
-                versions.insert(*slot_id, slot_version.saturating_add(1));
-            } else {
-                return Err(ClientError::NotConnected);
-            }
-
             if chunk_ack.accepted {
                 debug!("Chunk accepted by stackerdb: {chunk_ack:?}");
+                self.signer_db
+                    .set_latest_chunk_version(&signer_pk, slot_id.0, slot_version)
+                    .map_err(ClientError::SignerDBError)?;
```

Comment:
```suggestion
                    self.signer_db.set_latest_chunk_version(
                        &signer_pk,
                        slot_id.0,
                        slot_metadata.slot_version,
                    )?;
```

Reply:
Changed in https://github.com/stacks-network/stacks-core/pull/6106/commits/18b0f07b37e8cab1f36bcb11520525fd42ec6bf0